lib/constructor/index.js

```diff
@@ -389,20 +389,56 @@ Constructor.prototype.prepare = function prepare(spec, content, next) {
     if (err) return next(err);
 
     const pkgDir = path.join(outputPath, 'package');
-    async.series({
-      unpack: this.unpack.bind(this, { content, outputPath }),
-      install: this.install.bind(this, spec, pkgDir),
-      pack: this.pack.bind(this, pkgDir, tarPath),
-      upload: this.upload.bind(this, spec, tarPath)
-    }, (err) => {
-      if (err) return next(err);
-
-      next(null, { install: outputPath, tarball: tarPath });
+    //
+    // First see if this package has already been built for a different env, we
+    // should only build a single version once
+    //
+    this.checkAndDownload(spec, { tarball: tarPath }, next, () => {
```

Comment:
minor nitpick, might be worth renaming `tarPath` to `tarball` to use the shorthand